src/core_ocean/analysis_members/mpas_ocn_mixed_layer_depths.F

```diff
@@ -230,11 +232,18 @@ subroutine ocn_compute_mixed_layer_depths(domain, timeLevel, err)!{{{
          call mpas_pool_get_array(diagnosticsPool, 'potentialDensity', &
                         potentialDensity)
          call mpas_pool_get_array(diagnosticsPool, 'pressure', pressure)
+         call mpas_pool_get_array(forcingPool, 'landIcePressure', landIcePressure)
          call mpas_pool_get_array(diagnosticsPool, 'zMid', zMid)
          call mpas_pool_get_array(diagnosticsPool, 'zTop', zTop)
          call mpas_pool_get_array(meshPool, 'latCell', latCell)
          call mpas_pool_get_array(meshPool, 'lonCell', lonCell)
 
+         !If landice cavity remove land ice pressure to search for ML depth
+         if ( associated(landIcePressure) ) THEN
+           do iCell = 1,nCellsSolve
+             pressure(1,iCell) = pressure(1,iCell) - landIcePressure(iCell)
```

Comment:
@vanroekel, a couple of things.  First, it strikes me as maybe a bad idea to modify pressure and then change it back but maybe this is a common practice in MPAS-Ocean that I hadn't noticed.  It would seem better to me (though obviously more work) to use a scratch field. @mark-petersen, what do you think?

Second, I think `landIcePressure` needs to be subtracted from pressure at all levels, not just at `k = 1`.

Reply:
@xylar 
1) Using a separate field does make a lot of sense and removes the latter loop over pressure
2) Yep, you're right.  I was not fully understanding the code in the pressure computation.

I'll address both of these comments